src/Provider/AbstractProvider.php

```diff
@@ -21,11 +24,22 @@
  */
 abstract class AbstractProvider implements ProviderInterface
 {
-    protected $httpAdapter;
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory
+     */
+    private $httpMessageFactory;
 
-    public function __construct(HttpAdapterInterface $httpAdapter)
+    /**
+     * @param HttpClient $httpClient
+     */
+    public function __construct(HttpClient $httpClient = null)
```

Comment:
Sure, I'll remove the getter and setter and put the message factory in the constructor